docs/internals/frontends.rst

```diff
@@ -129,10 +131,78 @@ A debug log message::
     {"message": "35 self tests completed in 0.08 seconds",
      "type": "log_message", "created": 1488278449.5575905, "levelname": "DEBUG", "name": "borg.archiver"}
 
+Prompts
+-------
+
+Prompts assume a JSON form as well when the ``--log-json`` option is specified. Responses
+are still read verbatim from *stdin*, while prompts are JSON messages printed to *stderr*,
+just like log messages.
+
+Prompts use the *question_prompt*, *question_prompt_retry*, *question_invalid_answer*,
+*question_accepted_default*, *question_accepted_true*, *question_accepted_false* and
+*question_env_answer* types.
+
+The *message* property contains the same string displayed regularly in the same situation,
+while the *msgid* property may contain a msgid_, typically the name of the
+environment variable that can be used to override the prompt. It is the same for all JSON
+messages pertaining to the same prompt.
+
+The *is_prompt* boolean property distinguishes informational messages from prompts, it
+is true for *question_prompt* and *question_prompt_retry* types, otherwise it is false.
```

Comment:
hmm, is a boolean best here? it seems to be the kind of message, maybe more than 2 kinds could exist here.